lib/chef/knife/azure_server_create.rb

```diff
@@ -253,6 +253,14 @@ def load_winrm_deps
         :long => "--winrm-max-memoryPerShell MB",
         :description => "Set winrm max memory per shell in MB"
 
+
+      option :delete_chef_config,
+        :long => "--delete-chef-config",
+        :boolean => true,
+        :default => false,
+        :description => "Set this flag to delete chef configuration files during chef extension uninstall or update process it by default set to false"
```

Comment:
We should name this option in a way specific to the cloud-api use case, e.g. delete_chef_extension_config